controlplane/kubeadm/internal/control_plane.go

```diff
@@ -141,7 +141,9 @@ func (c *ControlPlane) InitialControlPlaneConfig() *bootstrapv1.KubeadmConfigSpe
 func (c *ControlPlane) JoinControlPlaneConfig() *bootstrapv1.KubeadmConfigSpec {
 	bootstrapSpec := c.KCP.Spec.KubeadmConfigSpec.DeepCopy()
 	bootstrapSpec.InitConfiguration = nil
-	bootstrapSpec.ClusterConfiguration = nil
+	// NOTE: For the joining we are preserving the ClusterConfiguration in order to determine if the
+	// cluster is using an external etcd in the kubeadm bootstrap provider (even if this is not required by kubeadm Join).
+	// TODO: Determine if this copy of cluster configuration can be used for rollouts (thus allowing to remove the annotation at machine level)
```

Comment:
After some thinking, it is not possible to use the kubeadm-config config map as a source of truth, because of the lack of conversion utils. So I'm preserving the ClusterConfiguration in the KubeadmConfig where it is immediately accessible from CABPK